arduino/WandCore/WandCore.ino

```diff
@@ -253,72 +257,114 @@ void loop()
-        debugI("touch is being held");
-        lastLogged = millis();
-      }
+  if (currentTouchEvent == SINGLE_TAP) {
+    debugI("Single Tap Just Happened");
+    if (wandMode == POWER) {                      // power mode, single tap == set light brightness based on "brightness" of color sensed
+      // TODO
+    }
+    else if (wandMode == COLOR) {                 // color mode, single tap == set color of light based on color sensed
+      // TODO
+    }
       
-      break;
+  } else if (currentTouchEvent == DOUBLE_TAP) {
+    debugI("Double Tap Just Happened");
+    // toggle the wand's mode
+    if (wandMode == POWER) {                      // power mode, double tap == switch to color mode
+      wandMode = COLOR;
+      incrementRGBColorloop();                    // resume cycling the onboard RGB through its colorloop
+      debugI("Switched to Color Mode");
+    } else {                                      // color mode, double tap == switch to power mode
+      wandMode = POWER;
+      setRGBLED(255, 255, 255);                   // solid white for power mode
+      debugI("Switched to Power Mode"); 
+    }
   }
-  
-  // Read IMU sensor
-  readIMU(imuData);
 
-  // Calculate sum of acceleration and compare to threshold
+  // Calculate sum of acceleration to compare to threshold
   float aSum = fabs(imuData[AX]) + fabs(imuData[AY]) + fabs(imuData[AZ]);
   float gSum = fabs(imuData[GX]) + fabs(imuData[GY]) + fabs(imuData[GZ]);
 
-  // Acceleration threshold reached, predict gesture
+  // Acceleration threshold reached, handle gesture
   if (aSum >= accelerationThresholdG || gSum >= gyroThreshold) {
+    // do the gesture predicting
     readAndPredictGesture(imuData, gestureConfidence);
     
-    if (gestureConfidence[GESTURE_FLICK] > 0.9) { // it's a flick, do flick things!
+    // read target light state
+    // only do this here as it's not needed elsewhere and this minimizes polling
+    // step 1 of 3 - prepare the raw string from the Hue library
+    String rawLightState;
+    rawLightState = myHue.getLightInfo(lightID);
+    int removeToIdx = rawLightState.indexOf("{") - 1;
+    rawLightState.remove(0, removeToIdx);
+
+    // step 2 of 3 - parse the JSON
+    StaticJsonDocument<112> filter;
+    JsonObject filter_state = filter.createNestedObject("state");
+    filter_state["hue"] = true;
+    filter_state["on"] = true;
+    filter_state["effect"] = true;
+    filter_state["bri"] = true;
+    filter_state["sat"] = true;
+    filter_state["ct"] = true;
+    
+    StaticJsonDocument<192> doc;
+    
+    DeserializationError error = deserializeJson(doc, rawLightState, DeserializationOption::Filter(filter));
+    
+    if (error) {
+      debugE("deserializeJson failed: %s", error.c_str());
+      return;
+    }
+
+    // step 3 of 3 - assign temporary state variables
+    JsonObject lightCurrentState = doc["state"];
+    long lightCurrentHue = lightCurrentState["hue"]; // 0(red)-65535(red); green = 21845 and blue = 43690

```

Comment:
You're going to need an HSL to RGB converter anyway if we want to shadow the color on the wand RGB :)

Reply:
yeahhh we're going to need to do some converting lol